src/Mobs/Monster.cpp

```diff
@@ -1294,3 +1401,60 @@ cMonster::eFamily cMonster::GetMobFamily(void) const
 {
 	return FamilyFromType(m_MobType);
 }
+
+
+
+
+
+void cMonster::LeashTo(cEntity * a_Entity, bool a_ShouldBroadcast)
+{
+	m_LeashedTo = a_Entity;
```

Comment:
You should probably add `ASSERT(m_LeashedTo == nullptr)`, so that when broken code in the future tries to call this while already leashed, the error is detected early and precisely.